redisvl/utils/vectorize/text/huggingface.py

```diff
@@ -74,95 +104,97 @@ def _initialize_client(self, model: str, **kwargs):
-
-        dtype = kwargs.pop("dtype", self.dtype)
-
         embedding = self._client.encode([text], **kwargs)[0]
-        return self._process_embedding(embedding.tolist(), as_buffer, dtype)
+        return embedding.tolist()
 
-    @deprecated_argument("dtype")
-    def embed_many(
-        self,
-        texts: List[str],
-        preprocess: Optional[Callable] = None,
-        batch_size: int = 10,
-        as_buffer: bool = False,
-        **kwargs,
-    ) -> Union[List[List[float]], List[bytes]]:
-        """Asynchronously embed many chunks of texts using the Hugging Face
-        sentence transformer.
+    def _embed_many(
+        self, texts: List[str], batch_size: int = 10, **kwargs
+    ) -> List[List[float]]:
+        """Generate vector embeddings for a batch of texts using the Hugging Face model.
 
         Args:
-            texts (List[str]): List of text chunks to embed.
-            preprocess (Optional[Callable], optional): Optional preprocessing
-                callable to perform before vectorization. Defaults to None.
-            batch_size (int, optional): Batch size of texts to use when creating
-                embeddings. Defaults to 10.
-            as_buffer (bool, optional): Whether to convert the raw embedding
-                to a byte string. Defaults to False.
+            texts: List of texts to embed
+            batch_size: Number of texts to process in each batch
+            **kwargs: Additional model-specific parameters
 
         Returns:
-            Union[List[List[float]], List[bytes]]: List of embeddings as lists of floats,
-            or as bytes objects if as_buffer=True
+            List[List[float]]: List of vector embeddings as lists of floats
 
         Raises:
-            TypeError: If the wrong input type is passed in for the test.
+            TypeError: If the input is not a list of strings
         """
         if not isinstance(texts, list):
             raise TypeError("Must pass in a list of str values to embed.")
         if len(texts) > 0 and not isinstance(texts[0], str):
             raise TypeError("Must pass in a list of str values to embed.")
 
-        dtype = kwargs.pop("dtype", self.dtype)
-
         embeddings: List = []
-        for batch in self.batchify(texts, batch_size, preprocess):
+        for batch in self.batchify(texts, batch_size, None):
             batch_embeddings = self._client.encode(batch, **kwargs)
-            embeddings.extend(
-                [
-                    self._process_embedding(embedding.tolist(), as_buffer, dtype)
-                    for embedding in batch_embeddings
-                ]
-            )
+            embeddings.extend([embedding.tolist() for embedding in batch_embeddings])
         return embeddings
 
+    async def _aembed(self, text: str, **kwargs) -> List[float]:
+        """Asynchronously generate a vector embedding for a single text.
+
+        Note: This implementation falls back to the synchronous version as
```

Comment:
This could be worth logging when these methods are called.

Reply:
done!